src/plugins/intel_cpu/src/emitters/plugin/riscv64/jit_eltwise_emitters.cpp

```diff
@@ -1600,6 +1600,99 @@ std::set<std::vector<element::Type>> jit_subtract_emitter::get_supported_precisi
+    prepare_table();
+}
+
+jit_tanh_emitter::jit_tanh_emitter(ov::intel_cpu::riscv64::jit_generator* host,
+                                   ov::intel_cpu::riscv64::cpu_isa_t host_isa,
+                                   ov::element::Type exec_prc)
+    : jit_emitter(host, host_isa, exec_prc),
+      jit_exp_emitter_(std::make_unique<jit_exp_emitter>(host, host_isa, exec_prc)) {
+    prepare_table();
+}
+
+size_t jit_tanh_emitter::get_inputs_num() const {
+    return 1;
+}
+
+size_t jit_tanh_emitter::aux_gprs_count() const {
+    OPENVINO_ASSERT(jit_exp_emitter_, "JIT Exp emitter is missed!");
+    return jit_exp_emitter_->aux_gprs_count();
+}
+
+size_t jit_tanh_emitter::aux_vecs_count() const {
+    OPENVINO_ASSERT(jit_exp_emitter_, "JIT Exp emitter is missed!");
+    // Need extra aux vectors for exp(x) and exp(-x)
+    return jit_exp_emitter_->aux_vecs_count() + 2;
+}
+
+size_t jit_tanh_emitter::aux_fp_gprs_count() const {
+    OPENVINO_ASSERT(jit_exp_emitter_, "JIT Exp emitter is missed!");
+    return std::max(jit_exp_emitter_->aux_fp_gprs_count(), 1lu);
+}
+
+void jit_tanh_emitter::emit_impl(const std::vector<size_t>& in_vec_idxs,
+                                 const std::vector<size_t>& out_vec_idxs) const {
+    if (host_isa_ == ov::intel_cpu::riscv64::cpu_isa_t::gv) {
+        emit_isa<ov::intel_cpu::riscv64::cpu_isa_t::gv>(in_vec_idxs, out_vec_idxs);
+    } else {
+        OPENVINO_THROW("Can't create jit eltwise kernel");
+    }
+}
+
+template <ov::intel_cpu::riscv64::cpu_isa_t isa>
+void jit_tanh_emitter::emit_isa(const std::vector<size_t>& in_vec_idxs,
+                                const std::vector<size_t>& out_vec_idxs) const {
+    auto src = VReg(in_vec_idxs[0]);
+    auto dst = VReg(out_vec_idxs[0]);
+    auto exp_pos = VReg(aux_vec_idxs[aux_vecs_count() - 2]);
+    auto exp_neg = VReg(aux_vec_idxs[aux_vecs_count() - 1]);
+
+    // exp(x)
+    {
+        std::vector<size_t> exp_src_idxs = {static_cast<size_t>(src.getIdx())};
+        std::vector<size_t> exp_dst_idxs = {static_cast<size_t>(exp_pos.getIdx())};
+        std::vector<size_t> exp_aux_vec_idxs(aux_vec_idxs.begin(), aux_vec_idxs.begin() + jit_exp_emitter_->aux_vecs_count());
+        jit_exp_emitter_->emit_code(exp_src_idxs, exp_dst_idxs, exp_aux_vec_idxs, aux_gpr_idxs, aux_fp_gpr_idxs);
+    }
+
+    // exp(-x)
+    {
+        // Negate src into exp_neg
+        h->vfneg_vv(exp_neg, src);
+        std::vector<size_t> exp_src_idxs = {static_cast<size_t>(exp_neg.getIdx())};
+        std::vector<size_t> exp_dst_idxs = {static_cast<size_t>(exp_neg.getIdx())};
+        std::vector<size_t> exp_aux_vec_idxs(aux_vec_idxs.begin(), aux_vec_idxs.begin() + jit_exp_emitter_->aux_vecs_count());
+        jit_exp_emitter_->emit_code(exp_src_idxs, exp_dst_idxs, exp_aux_vec_idxs, aux_gpr_idxs, aux_fp_gpr_idxs);
+    }
+
+    // tanh(x) = (exp(x) - exp(-x)) / (exp(x) + exp(-x))
+    h->vfsub_vv(dst, exp_pos, exp_neg); // numerator
```

Comment:
`Exp` is heavy activation which has high performance cost. Current implementation emits two `jit_exp_emitter` increasing cost of `jit_tanh_emitter` twice. Can we call only one `jit_exp_emitter_->emit_code()` here (or even another JIT emitter)?

Please take a look at the following formulas:
- tanh:
<img width="445" height="42" alt="image" src="https://github.com/user-attachments/assets/ae348715-4400-498e-8135-4092d3c8d1ad" />

- sigmoid:
<img width="145" height="45" alt="image" src="https://github.com/user-attachments/assets/f741952b-3dbd-49fe-8734-ab6c6cd03c0f" />

I believe it might be helpful for you 😊

Reply:
The implementation has not been changed after my comment - please double-check